libs/geoservices/covid/src/lib/pages/maps/components/lockdown-map/lockdown-map.component.ts

```diff
@@ -0,0 +1,258 @@
+import { Component, OnInit } from '@angular/core';
+import { HttpClient } from '@angular/common/http';
+
+import { forkJoin, BehaviorSubject } from 'rxjs';
+import { take } from 'rxjs/operators';
+
+import { MapboxMapService } from '@tamu-gisc/maps/mapbox';
+
+@Component({
+  selector: 'tamu-gisc-lockdown-map',
+  templateUrl: './lockdown-map.component.html',
+  styleUrls: ['./lockdown-map.component.scss']
+})
+export class LockdownMapComponent implements OnInit {
+  constructor(private mapService: MapboxMapService, private http: HttpClient) {}
+  public unformattedData;
+  public statData: Array<StatRecord>;
+  public lockdownButtonToggle = false;
+  public infoBoxModel: BehaviorSubject<IInfoBox> = new BehaviorSubject(undefined);
+  public countyBoxModel: BehaviorSubject<CountyStateBox> = new BehaviorSubject(undefined);
+  public ngOnInit(): void {
+    this.mapService.loaded.subscribe((map) => {
+      this.getStats();
+      const zoomThreshold = 3;
+
+      map.addSource('county-lines', {
+        type: 'vector',
+        url: 'mapbox://gsepulveda96.countylines'
+      });
+
+      map.addSource('state-lines', {
+        type: 'vector',
+        url: 'mapbox://gsepulveda96.statelines'
+      });
+
+      map.addLayer(
+        {
+          id: 'counties-lockdowns',
+          type: 'fill',
+          source: 'county-lines',
+          layout: {
+            visibility: 'none'
+          },
+          'source-layer': 'county-lines',
+          paint: {
+            'fill-outline-color': '#303030'
+          }
+        },
+        'road-motorway-trunk'
+      );
+
+      map.addLayer(
+        {
+          id: 'counties-claims',
+          type: 'fill',
+          source: 'county-lines',
+          'source-layer': 'county-lines',
+          paint: {
+            'fill-outline-color': '#FFFFFF'
+          }
+        },
+        'counties-lockdowns'
+      );
+
+      map.addLayer(
+        {
+          id: 'statelines',
+          type: 'line',
+          source: 'state-lines',
+          'source-layer': 'state-lines',
+          paint: {
+            'line-color': '#000000',
+            'line-width': 1
+          }
+        },
+        'state-label'
+      );
+
+      map.on('mousemove', 'counties-lockdowns', (e) => {
```

Comment:
Not a showstopper: If you can get the mapbox-gl types working on this library, if they aren't there yet already, try and type all of the things. Event objects can be particularly frustrating to work with them since there's so many of them and in this case should not be confused with native mouse events (?).

Reply:
set these as County Type objects